File: Data Structures & Algorithms/top-k-elements-in-list/submission-0.js
class Solution {
    /**
     * @param {number[]} nums
     * @param {number} k
     * @return {number[]}
     */
    topKFrequent(nums, k) {
        let map = new Map();
        let result = [];
        let count = 0 ;
        for(let i = 0; i < nums.length; i++) {
            if(map.has(nums[i])) {
                map.set(nums[i], map.get(nums[i]) + 1);
            } else {
                map.set(nums[i], 1);
            }
        }
        let sortedMap = new Map([...map.entries()].sort((a,b) => b[1] - a[1]));

        for(let [key,value] of sortedMap) {
            if(count < k) {
                result.push(key);
                count++;
            }
        }
        return result;

    }
}

File: Data Structures & Algorithms/products-of-array-discluding-self/submission-0.py
class Solution:
    def productExceptSelf(self, nums: List[int]) -> List[int]:
       r, l = 1, 1
       result = [1] * len(nums)
       n = len(nums)
       for i in range(n):
        result[i] = result[i] * l
        l = l * nums[i]
       for i in range(n-1, -1, -1):
        result[i] = result[i] * r
        r = r * nums[i]
       return result
    
    
        
        
 
        
            
        